### PR TITLE
Fixes #34238 - resolve migration issues after core migration cleanup

### PR DESCRIPTION
**Pull request by lzap**
This reverts commit 8145219c02a0cad97a15aee727e56f3e8867c76a.

https://community.theforeman.org/t/luna-nightly-rpm-pipeline-322-failed/26760

### CODE REVIEW

File: db/migrate/20140522185700_change_templatekind_to_bootdisk.rb [hunk 1]
```diff
@@ -1,14 +1,22 @@
 # frozen_string_literal: true
 
 class ChangeTemplatekindToBootdisk < ActiveRecord::Migration[4.2]
+  class FakeConfigTemplate < ApplicationRecord
+    self.table_name = if ActiveRecord::Base.connection.migration_context.get_all_versions.include?(20_150_514_072_626)
+                        'templates'
+                      else
+                        'config_templates'
+                      end
```
Comment (evgeni) [inline db/migrate/20140522185700_change_templatekind_to_bootdisk.rb:9]: I think this is wrong.

https://github.com/theforeman/foreman/commit/9851420902165fc23d0ea1f6d53922a5f6a37c6a dropped the old `20150514114044_migrate_ptables_to_templates.rb` migration, so the `include?` will be `false`, and you'd end up with `config_templates` name, not `templaes` for the table name.

Comment (evgeni) [reply db/migrate/20140522185700_change_templatekind_to_bootdisk.rb:9]: Hmm, no, it dropped only parts of that migration. So it actually could work.

@m-bucher could you have a look, was the removal in #110 purely cleanup, or was there a bug?

Comment (m-bucher) [reply db/migrate/20140522185700_change_templatekind_to_bootdisk.rb:9]: IIRC we had problems during migration in certain cases. Though I am not sure in which scenarios specifically :thinking:

Comment (lzap) [reply db/migrate/20140522185700_change_templatekind_to_bootdisk.rb:9]: Hmm but the statement queries `db/migrate/20150514072626_add_type_to_config_template.rb` and not `20150514114044_migrate_ptables_to_templates.rb`. The former migration is still present even today in develop.

Comment (lzap) [reply db/migrate/20140522185700_change_templatekind_to_bootdisk.rb:9]: Yeah, I did not notice the conversation after the initial comment...

Anyway, isn't safer to simply test for a class constant presence? Something like `if defined? ConfigTemplates` ?

Comment (lzap) [reply db/migrate/20140522185700_change_templatekind_to_bootdisk.rb]: I pushed another commit that feels safer to me, what you guys think? That does not depend on any migration numbers or whatever, it simply tests if older class constant exists, if it does not it is named the new way.

Comment (evgeni) [reply db/migrate/20140522185700_change_templatekind_to_bootdisk.rb]: > Hmm but the statement queries `db/migrate/20150514072626_add_type_to_config_template.rb` and not `20150514114044_migrate_ptables_to_templates.rb`. The former migration is still present even today in develop.

Heh. I failed on Ctrl+F and this silly notation of the migration version. You're right.

**Comment by lzap** [comment]
Note: Squash on merge into "Fixes #34238 - resolve migration issues after core migration cleanup"

**Comment by lzap** [comment]
So what do you guys say? Merge and new release to fix nightlies? Or other concerns?

**Comment by lzap** [comment]
* https://github.com/theforeman/foreman-packaging/pull/7454
* https://github.com/theforeman/foreman-packaging/pull/7455

**Pull request merged**